tests/test_tacacs_plus.py

```diff
@@ -167,13 +177,137 @@ def test_auth_reply_unpack_server_msg():
     assert reply.human_status == 'GETPASS'
 
 
+# test authorization objects
+def test_authorization_start_pack():
+    username = 'user123'
+    packed = TACACSAuthorizationStart(
+        username, TAC_PLUS_AUTHEN_METH_TACACSPLUS,
+        TAC_PLUS_PRIV_LVL_MIN, TAC_PLUS_AUTHEN_TYPE_ASCII,
+        [b"service=shell", b"cmd=show", b"cmdargs=version"],
+    ).packed
+    assert packed == six.b(
+        '\x06'              # tacacs_plus.flags.TAC_PLUS_AUTHEN_METH_TACACSPLUS \
+        '\x00'              # tacacs_plus.flags.TAC_PLUS_PRIV_LVL_MIN \
+        '\x01'              # tacacs_plus.flags.TAC_PLUS_AUTHEN_TYPE_ASCII \
+        '\x01'              # tacacs_plus.flags.TAC_PLUS_AUTHEN_SVC_LOGIN \
+        '\x07'              # username_len == 7 \
+        '\x0b'              # port_len == 11 \
+        '\x0d'              # rem_addr_len == 13 \
+        '\x03'              # arg_len == 3
+        '\x0d'              # arg_1_len == 13 (service=shell) \
+        '\x08'              # arg_2_len == 8 (cmd=show) \
+        '\x0f'              # arg_3_len == 14 (cmdargs=version) \
+        'user123'           # username \
+        'python_tty0'       # port \
+        'python_device'     # rem_addr \
+        'service=shell'     # arg_1 (service=shell) \
+        'cmd=show'          # arg_2 (cmd=show) \
+        'cmdargs=version'   # arg_3 (cmdargs=version)
+    )
+
+
+@pytest.mark.parametrize('raw, state, human_status', [
+    (six.b('\x01\x00\x00\x00\x00\x00'), 'valid', 'PASS'),
+    (six.b('\x02\x00\x00\x00\x00\x00'), 'reply', 'REPL'),
+    (six.b('\x10\x00\x00\x00\x00\x00'), 'invalid', 'FAIL',),
+    (six.b('\x11\x00\x00\x00\x00\x00'), 'error', 'ERROR'),
+    (six.b('\x21\x00\x00\x00\x00\x00'), 'follow', 'FOLLOW'),
+])
+def test_authorization_reply_unpack(raw, state, human_status):
```

Comment:
Ahahaha you are more than welcome! Thanks for starting this, I really appreciate the effort.